packages/core/src/interpolateColors.ts

```diff
@@ -0,0 +1,740 @@
+			return color;
+		}
+		return null;
+	}
+
+	if (typeof color !== 'string') {
+		return null;
+	}
+
+	const matchers = getMatchers();
+
+	let match: RegExpExecArray | null;
+
+	// Ordered based on occurrences on Facebook codebase
+	if (matchers.hex6) {
+		if ((match = matchers.hex6.exec(color))) {
+			return Number.parseInt(match[1] + 'ff', 16) >>> 0;
+		}
+	}
+
+	if (names[color] !== undefined) {
+		return names[color];
+	}
+	if (matchers.rgb) {
+		if ((match = matchers.rgb.exec(color))) {
+			return (
+				// b
+				((parse255(match[1]) << 24) | // r
+				(parse255(match[2]) << 16) | // g
+					(parse255(match[3]) << 8) |
+					0x000000ff) >>> // a
+				0
+			);
+		}
+	}
+
+	if (matchers.rgba) {
+		if ((match = matchers.rgba.exec(color))) {
+			return (
+				// b
+				((parse255(match[1]) << 24) | // r
+				(parse255(match[2]) << 16) | // g
+					(parse255(match[3]) << 8) |
+					parse1(match[4])) >>> // a
+				0
+			);
+		}
+	}
+
+	if (matchers.hex3) {
+		if ((match = matchers.hex3.exec(color))) {
+			return (
+				Number.parseInt(
+					match[1] +
+					match[1] + // r
+					match[2] +
+					match[2] + // g
+					match[3] +
+					match[3] + // b
+						'ff', // a
+					16
+				) >>> 0
+			);
+		}
+	}
+
+	// https://drafts.csswg.org/css-color-4/#hex-notation
+	if (matchers.hex8) {
+		if ((match = matchers.hex8.exec(color))) {
+			return Number.parseInt(match[1], 16) >>> 0;
+		}
+	}
+
+	if (matchers.hex4) {
+		if ((match = matchers.hex4.exec(color))) {
+			return (
+				Number.parseInt(
+					match[1] +
+					match[1] + // r
+					match[2] +
+					match[2] + // g
+					match[3] +
+					match[3] + // b
+						match[4] +
+						match[4], // a
+					16
+				) >>> 0
+			);
+		}
+	}
+
+	if (matchers.hsl) {
+		if ((match = matchers.hsl.exec(color))) {
+			return (
+				(hslToRgb(
+					parse360(match[1]), // h
+					parsePercentage(match[2]), // s
+					parsePercentage(match[3]) // l
+				) |
+					0x000000ff) >>> // a
+				0
+			);
+		}
+	}
+
+	if (matchers.hsla) {
+		if ((match = matchers.hsla.exec(color))) {
+			return (
+				(hslToRgb(
+					parse360(match[1]), // h
+					parsePercentage(match[2]), // s
+					parsePercentage(match[3]) // l
+				) |
+					parse1(match[4])) >>> // a
+				0
+			);
+		}
+	}
+
+	return null;
+}
+
+export const opacity = (c: number): number => {
+	return ((c >> 24) & 255) / 255;
+};
+
+export const red = (c: number): number => {
+	return (c >> 16) & 255;
+};
+
+export const green = (c: number): number => {
+	return (c >> 8) & 255;
+};
+
+export const blue = (c: number): number => {
+	return c & 255;
+};
+
+export const rgbaColor = (
+	r: number,
+	g: number,
+	b: number,
+	alpha = 1
+): string => {
+	return `rgba(${r}, ${g}, ${b}, ${alpha})`;
+};
+
+/* accepts parameters
+ * r  Object = {r:x, g:y, b:z}
+ * OR
+ * r, g, b
+ * 0 <= r, g, b <= 255
+ * returns 0 <= h, s, v <= 1
+ */
+
+/* eslint no-redeclare: 0 */
+function RGBtoHSV(rgb: RBG): HSV;
+function RGBtoHSV(r: number, g: number, b: number): HSV;
```

Comment:
really strange that functions are declared multiple times. can you investigate if we need this?